File: Structural-Patterns/Adapter/C#/Adapter/Adapters/GPayAdapter.cs
public class GPayAdapter : IUPIPaymentGateway{

    private GPayAPI gPayAPI = new GPayAPI();

    public UPIPaymentResponse InitiatePayment(UPIPaymentRequest request) {
        Console.WriteLine("--> Using GooglePayAdapter to translate our request...");
        String transactionId = gPayAPI.startPayment(request.GetOrderId());

        PaymentStatus status = PaymentStatus.SUCCESS;

        return new UPIPaymentResponse(transactionId, status);
    }

    public PaymentStatus CheckStatus(String transactionId) {
        String status = gPayAPI.fetchStatus(transactionId);
        return PaymentStatus.SUCCESS;
    }
    
}

File: Structural-Patterns/Adapter/Java/Adapters/GPayAdapter.java
package Adapters;

import Adaptee.GPayAPI;
import Implementations.PaymentStatus;
import Implementations.UPIPaymentRequest;
import Implementations.UPIPaymentResponse;
import Interfaces.IUPIPaymentGateway;

public class GPayAdapter implements IUPIPaymentGateway{

    private final GPayAPI gPayAPI = new GPayAPI();

    @Override
    public UPIPaymentResponse initiatePayment(UPIPaymentRequest request) {
        System.out.println("--> Using GooglePayAdapter to translate our request...");
        String transactionId = gPayAPI.startPayment(request.getOrderId());

        PaymentStatus status = PaymentStatus.valueOf(gPayAPI.fetchStatus(transactionId));

        return new UPIPaymentResponse(transactionId, status);
    }

    @Override
    public PaymentStatus checkStatus(String transactionId) {
        String status = gPayAPI.fetchStatus(transactionId);
        return PaymentStatus.valueOf(status);
    }
    
}
